opus/application/static_media/js/opus.js

```diff
@@ -44,7 +44,23 @@ var opus = {
     // avoiding race conditions in ajax calls
     lastAllNormalizeRequestNo: 0,
     lastResultCountRequestNo: 0,
-    waitingForAllNormalizedAPI: false,
+
+    // An object stores the input (slug) & boolean value to indicate whether the normalize input
+    // API (for all fields) is in progress:
+    // Key: input slug that triggers the normalize input API call. (Use opus.allSlug for API from getResultCount)
+    // Value: true & false to indicate whether the API is in progress.
+    normalizeInputForAllFieldsInProgress: {},
```

Comment:
Rather than allowing the field to be true or false, which gives complicated code like `isAnyNormalizeInputInProgress` below, would it be better to have a slug value be present or not present? Then to check if any normalize is in progress, you can just see if this object has the slug as a property.

Are there any properties? `Object.keys(normalizeInputForAllFieldsInProgress).length > 0`
Is there a particular property? `if (normalizeInputForAllFieldsInProgress[slug])`
Set a particular property? `normalizeInputForAllFieldsInProgress[slug] = true;`
Remove a particular property? `delete normalizeInputForAllFieldsInProgress[slug];`

I'm not sure if this is better or cleaner - I'll let you decide.